src/sections/trade/sections/bonds/MyActiveBonds.tsx

```diff
@@ -1,39 +1,58 @@
 import { BondsTable } from "./table/BondsTable"
 import { Skeleton } from "./table/skeleton/Skeleton"
-import { useQuery } from "@tanstack/react-query"
-import { Bond } from "./table/BondsTable.utils"
-
-export const MyActiveBonds = () => {
-  // TODO: replace mocked values with fetched data
-  const { data, isLoading } = useQuery(["dummy"], () => {
-    return new Promise<Bond[]>((resolve) =>
-      setTimeout(() => {
-        resolve([
-          {
-            assetId: "1",
-            maturity: 1693314639000,
-            balance: "300 845 HDX",
-            price: "1$",
-          },
-          {
-            assetId: "2",
-            maturity: 1693314639000,
-            balance: "300 845 HDX",
-            price: "1$",
-          },
-        ])
-      }, 2000),
-    )
-  })
-
-  const props = {
-    title: "My Active Bonds",
+import { useTokensBalances } from "api/balances"
+import { useAccountStore } from "state/store"
+import { Bond } from "api/bonds"
+import { pluck } from "utils/rx"
+import { BondTableItem } from "./table/BondsTable.utils"
+import { useTranslation } from "react-i18next"
+import { u32 } from "@polkadot/types-codec"
+import { u8 } from "@polkadot/types"
+
+interface Props {
+  bonds: Bond[]
+  metas: { id: string; decimals: u32 | u8 }[]
+  isLoading?: boolean
+}
+
+export const MyActiveBonds = ({ bonds, metas, ...props }: Props) => {
+  const { t } = useTranslation()
+  const { account } = useAccountStore()
+  const balances = useTokensBalances(pluck("id", bonds), account?.address)
+
+  const isLoading =
+    pluck("isLoading", balances).some(Boolean) || props.isLoading
+
+  const tableProps = {
+    title: t("bonds.table.title"),
     showTransactions: false,
   }
 
   if (isLoading) {
-    return <Skeleton {...props} />
+    return <Skeleton {...tableProps} />
   }
 
-  return <BondsTable {...props} data={data ?? []} />
+  const bondMap = new Map(bonds.map((bond) => [bond.id, bond]))
+  const metaMap = new Map(metas.map((meta) => [meta.id, meta]))
+
+  const data = balances
+    .filter((balance) => balance.data?.assetId)
```

Comment:
Could you use reduce fn instead of filter+map?

Reply:
Possibly yes, I just find it more readable to do it like this. It's probably also a bit more efficient